tag-state/fix-tag-state.py

```diff
@@ -245,19 +249,24 @@ def pretty_printable_dict(d: dict) -> str:
                 )
                 # if creation and annotation worked, push the new manifest
                 if (
-                    annotate_manifest_arm.returncode is 0
-                    and annotate_manifest_amd.returncode is 0
+                    annotate_manifest_arm.returncode == 0
+                    and annotate_manifest_amd.returncode == 0
                 ):
                     # rebuilds aren't a factor now and the tags are semver sorted, so the last writer is the right one
                     major_manifest_name = f"gcr.io/ironcore-images/{image_name}:{major}"
                     minor_manifest_name = (
                         f"gcr.io/ironcore-images/{image_name}:{major}.{minor}"
                     )
+                    # you can't create a manifest to move it if you've already created it locally without amending
+                    # unfortunately if you _do_ amend it just adds the new images to the mafest list instead of moving the tag.
+                    # we try to get around this by purging on push so each create is "new"
+                    # TODO: this isn't currently working, so major/minors need to be manually checked and moved afterwards
```

Comment:
Could you do a `docker manifest rm` before you create it?

Reply:
I had tried that before this, but it had the same result as it currently does (won't push without amend). It seems like no matter what I do it either: 

a) results in "won't push without amend"
b) "amending" a giant list of images into one manifest